### PR TITLE
openai-whisper: 20230314 -> 20230918

### PR DESCRIPTION
**Pull request by MayNiklas**
## Description of changes

Changelog:
https://github.com/openai/whisper/blob/v20230918/CHANGELOG.md

Relevant changes for us:
instead of using `ffmpeg-python`, the project now calls `ffmpeg` directly.

I only tested the CPU version. The CUDA build should be ready in ~12 hours.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/openai-whisper/default.nix [hunk 1]
```diff
@@ -43,12 +42,13 @@ buildPythonPackage rec {
     })
   ];
 
+  buildInputs = [ ffmpeg ];
```
Comment (mweinelt) [inline pkgs/development/python-modules/openai-whisper/default.nix:45]: Why is this required? And can we use ffmpeg-headless instead? It has a much smaller closure size.

Comment (MayNiklas) [reply pkgs/development/python-modules/openai-whisper/default.nix:45]: You're right - it is not! I was confused by the release notes. I'm going to test ffmpeg-headless.

Comment (MayNiklas) [reply pkgs/development/python-modules/openai-whisper/default.nix:45]: `ffmpeg-headless`seems to work!

**Comment by MayNiklas** [comment]
Also worked with CUDA 👍

**Review by mweinelt — APPROVED** [review]

**Comment by mweinelt** [comment]
@ofborg build openai-whisper

File: pkgs/development/python-modules/openai-whisper/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/python-modules/openai-whisper/default.nix]: Closure size reduced from 2.16 to 1.85 GiB. Over half of the rest is Pytorch.

**Comment by mweinelt** [comment]
Not sure if you care for today's Tagesschau, but this is how it starts.

```
[00:00.000 --> 00:08.000]  Hier ist das erste deutsche Fernsehen mit der Tagesschau.
[00:08.000 --> 00:18.000]  Heute im Studio, Konstantin Schreiber.
[00:18.000 --> 00:21.000]  Guten Abend, meine Damen und Herren.
[00:21.000 --> 00:23.000]  Ich begrüße Sie zur Tagesschau.
[00:23.000 --> 00:27.000]  Ein Krisengipfel im Kanzleramt hat sich heute mit der Frage beschäftigt,
[00:27.000 --> 00:31.000]  wie wieder mehr bezahlbarer Wohnraum geschaffen werden kann.
[00:31.000 --> 00:35.000]  Eingeladen waren Vertreterinnen und Vertreter unterschiedlicher Branchen,
[00:35.000 --> 00:37.000]  Verbände sowie Gewerkschaften.
[00:37.000 --> 00:42.000]  Höhere Zinsen und teurere Baustoffe, aber auch geplante Klimaschutzvorgaben
[00:42.000 --> 00:45.000]  haben die Branche in eine schwere Krise geführt.
[00:45.000 --> 00:51.000]  Der Mangel an Wohnraum sorgt vor allem in Ballungsräumen für explodierende Mieten.
[00:51.000 --> 00:55.000]  Jürgen Lenz schaut sich eine leer stehende Wohnung in Ludwigshafen an.
[00:55.000 --> 00:58.000]  Der Bauingenieur arbeitet für eine Wohnungsbaugesellschaft,
[00:58.000 --> 01:00.000]  der die Immobilie gehört.
[01:00.000 --> 01:02.000]  Hier wird mit Gas geheizt.
[01:02.000 --> 01:05.000]  Der Einbau einer Wärmepumpe in dieses Gebäude
[01:05.000 --> 01:10.000]  würde allein für diese Wohnung 127.000 Euro kosten.
[01:10.000 --> 01:13.000]  Komplett alles auszutauschen oder eben auch Fassadendämmung,
[01:13.000 --> 01:15.000]  die Fenster komplett zu machen.
[01:15.000 --> 01:18.000]  Das ist eigentlich wissenschaftlich nicht mehr darstellbar.
[01:18.000 --> 01:22.000]  In einem anderen Stadtteil errichtet die Gesellschaft gerade einen Neubau.
[...]
```
